build.js

```diff
@@ -16,81 +16,168 @@ function getDirname () {
 
 // it's basically the old file with /source/ replaced with /build/
 function getBuildFilePathFromSourceFilePath (sourceFilePath) {
-	// note that replace only replaces the first occurence if the first argument
+	// note that replace only replaces the first occurrence if the first argument
 	//  is a string
 	return sourceFilePath.replace(
 		getDirname() + '/source/',
 		getDirname() + '/build/'
 	)
 }
 
+function getFileExtension (filePath) {
+	return filePath.split('.').pop()
+}
+
 function isHtmlFile (filePath) {
-	return filePath.slice(-5) === '.html'
+	return getFileExtension(filePath) === 'html'
 }
 
-// create the target build/ folder if it doesn't exist yet
-if (!fs.existsSync('./build/')){
-	fs.mkdirSync('./build/')
-	console.log('Created build folder')
+function isCssFile (filePath) {
+	return getFileExtension(filePath) === 'css'
+}
+
+function makeFolder (folderPath) {
+	// console.log('Preparing folder', folderPath)
+	// create if it doesn't exists yet
+	if (!fs.existsSync(folderPath)) {
+		fs.mkdirSync(folderPath)
+		console.log('Created folder', folderPath)
+	}
+}
+
+// creates a dictionary of (filename: sourceFile) for Handlebars
+function makeComponentDictionary(globPattern) {
+	console.log('Looking for components matching: ' + globPattern)
+
+	const componentSourceFiles = glob.sync(globPattern)
+	const components = {}
+	for (const componentSourceFile of componentSourceFiles) {
+		const componentName = componentSourceFile
+			.split('/').pop() // get the actual filename
+			.replace(/\.html$/, '') // remove .html at the end
+	
+		console.log('Found component', componentName)
+
+		// parse component
+		const fileString = fs.readFileSync(componentSourceFile, "utf8")
+		if (fileString.indexOf('TEMPORARY_OPENING_SIGNATURE ') == 0) {
```

Comment:
maybe we can replace these with variables

Reply:
signatures have been turned into constants defined on the top of the file
```
// syntax for compound components
const OPENING_SIGNATURE = 'TEMPORARY_OPENING_SIGNATURE ' // notice the space on end
const CLOSING_SIGNATURE = ' TEMPORARY_CLOSING_SIGNATURE' // notice the space in front
```